boa/src/context.rs

```diff
@@ -380,10 +392,26 @@ impl Context {
         hint: PreferredType,
     ) -> Result<Value> {
         // 1. Assert: Type(O) is Object.
+        // TODO: #776 this should also accept Type::Function
         debug_assert!(o.get_type() == Type::Object);
         // 2. Assert: Type(hint) is String and its value is either "string" or "number".
         debug_assert!(hint == PreferredType::String || hint == PreferredType::Number);
 
+        // Diverge from the spec here to make sure we aren't going to overflow the stack by converting
+        // a recursive structure
+        // We can follow v8 & SpiderMonkey's lead and return a default value for the hint in this situation
+        // (see https://repl.it/repls/IvoryCircularCertification#index.js)
+        let obj = o.as_gc_object().unwrap(); // UNWRAP: Asserted type above
```

Comment:
Mostly stylistic, but I think using `.expect("Asserted type above")` is preferable to `.unwrap()` with a comment.

Reply:
Unwrap isn't necessary anymore because the signature now takes a GcObject directly.